src/bindgen/ir/function.rs

```diff
@@ -46,10 +47,16 @@ impl Function {
     ) -> Result<Function, String> {
         let mut args = sig.inputs.iter().try_skip_map(|x| x.as_ident_and_type())?;
 
+        let mut never_return = false;
         let mut ret = match sig.output {
             syn::ReturnType::Default => Type::Primitive(PrimitiveType::Void),
             syn::ReturnType::Type(_, ref ty) => {
-                Type::load(ty)?.unwrap_or_else(|| Type::Primitive(PrimitiveType::Void))
+                if let syn::Type::Never(_) = ty as &syn::Type {
```

Comment:
Do you really need the `as &syn::Type`?

Reply:
I guess the best thing I can do is to replace `ty as &syn::Type` with `ty.as_ref()`; will do that